apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx

```diff
@@ -115,34 +127,307 @@ const PostgrestConfig = () => {
         }
       }) ?? []
 
-  const isGraphqlExtensionEnabled =
-    (extensions ?? []).find((ext) => ext.name === 'pg_graphql')?.installed_version !== null
+  function resetForm(values?: Partial<z.infer<typeof formSchema>>) {
+    const dbSchema =
+      config?.db_schema && config?.db_schema ? config.db_schema.replace(/ /g, '').split(',') : []
+
+    if (values?.enableDataApi && dbSchema.length === 0) {
+      dbSchema.push('public', 'storage')
+    }
+
+    const defaultValues = {
+      dbSchema: values?.dbSchema ?? dbSchema,
+      maxRows: values?.maxRows ?? config?.max_rows,
+      dbExtraSearchPath: values?.dbExtraSearchPath ?? config?.db_extra_search_path,
+      dbPool: values?.dbPool ?? config?.db_pool,
+      enableDataApi: values?.enableDataApi,
+    }
 
-  function resetForm() {
     if (config) {
       form.reset({
-        dbSchema: config.db_schema ? config.db_schema.replace(/ /g, '').split(',') : [],
-        maxRows: config.max_rows,
-        dbExtraSearchPath: config.db_extra_search_path,
-        dbPool: config.db_pool,
+        ...defaultValues,
       })
     }
   }
 
+  const onSubmit = async (values: z.infer<typeof formSchema>) => {
+    if (!projectRef) return console.error('Project ref is required') // is this needed ?
```

Comment:
the console.error is technically not needed, just minimally the return
its just the type checking from the mutation that expects projectRef to not be undefined

Reply:
this was here before hand, it just got moved.
can we do in another PR so we don't add another point of failure ?